tools/structure/src/main.py

```diff
@@ -70,46 +73,72 @@ def run(
+                embedding_instance_id=tool_settings[SettingsKeys.EMBEDDING],
+                vector_db_instance_id=tool_settings[SettingsKeys.VECTOR_DB],
+                x2text_instance_id=tool_settings[SettingsKeys.X2TEXT_ADAPTER],
                 file_path=input_file,
                 file_hash=file_hash,
                 chunk_size=tool_settings[SettingsKeys.CHUNK_SIZE],
                 chunk_overlap=tool_settings[SettingsKeys.CHUNK_OVERLAP],
+                output_file_path=tool_data_dir / SettingsKeys.EXTRACT,
                 reindex=True,
             )
-            self.stream_log("Fetching response for single pass extraction...")
+            if summarize_as_source:
+                self._summarize_and_index(
+                    tool_id=tool_id,
+                    tool_settings=tool_settings,
+                    tool_data_dir=tool_data_dir,
+                    responder=responder,
+                    outputs=outputs,
+                    index=index,
+                )
+            self.stream_log("Fetching response for single pass extraction")
             prompt_service_resp = responder.single_pass_extraction(payload=payload)
         else:
-            self.stream_log("Indexing document...")
             try:
+                # To reindex even if file is already
+                # indexed to get the output in required path
+                reindex = True
                 for output in outputs:
-                    tool_index.index_file(
+                    index.index_file(
                         tool_id=tool_metadata[SettingsKeys.TOOL_ID],
-                        embedding_type=output[SettingsKeys.EMBEDDING],
-                        vector_db=output[SettingsKeys.VECTOR_DB],
-                        x2text_adapter=output[SettingsKeys.X2TEXT_ADAPTER],
+                        embedding_instance_id=output[SettingsKeys.EMBEDDING],
+                        vector_db_instance_id=output[SettingsKeys.VECTOR_DB],
+                        x2text_instance_id=output[SettingsKeys.X2TEXT_ADAPTER],
                         file_path=input_file,
                         file_hash=file_hash,
                         chunk_size=output[SettingsKeys.CHUNK_SIZE],
                         chunk_overlap=output[SettingsKeys.CHUNK_OVERLAP],
-                        reindex=output[SettingsKeys.REINDEX],
+                        output_file_path=tool_data_dir / SettingsKeys.EXTRACT,
+                        reindex=reindex,
                     )
+                    if summarize_as_source:
+                        self._summarize_and_index(
+                            tool_id=tool_id,
+                            tool_settings=tool_settings,
+                            tool_data_dir=tool_data_dir,
+                            responder=responder,
+                            outputs=output,
+                            index=index,
+                        )
+                        break
```

Comment:
NIT: Add a comment on why we're breaking here for summary

Reply:
Sure. It is the same reason as the comment above the for loop.